scripts/download.sh

```diff
@@ -52,24 +52,15 @@ mkdir -p beta-amd64/
 tar xzvf beta-amd64.tar.gz -C beta-amd64/
 chmod +x beta-amd64/buildkite-agent
 
+rm -rf agent-master/
+download https://github.com/buildkite/agent/archive/master.zip master.zip
+unzip master.zip
+cd agent-master/
+docker build --tag agent-master .
+docker run --rm -v "$PWD:/go/src/github.com/buildkite/agent" agent-master scripts/utils/build-binary.sh "linux" "amd64" "edge"
+cd ../
```

Comment:
Could this perhaps consume the compiled release from https://buildkite.com/install.sh / https://buildkite.com/agent/releases/latest?platform=linux&arch=amd64 instead? Or should we rebuild it here?